chia/wallet/wallet_state_manager.py

```diff
@@ -1159,19 +1176,38 @@ async def new_coin_state(
                     coin_name = coin_added.name()
                     existing = await self.coin_store.get_coin_record(coin_name)
                     if existing is None:
-                        await self.coin_added(
-                            coin_added,
-                            uint32(coin_state.spent_height),
-                            [],
-                            pool_wallet.id(),
-                            WalletType(pool_wallet.type()),
-                            peer,
-                            coin_name,
+                        deferred_coins_added.append(
+                            DeferredCoinAdd(
+                                coin_added,
+                                uint32(coin_state.spent_height),
+                                [],
+                                pool_wallet.id(),
+                                WalletType(pool_wallet.type()),
+                                peer,
+                                coin_name,
+                            )
                         )
                     await self.add_interested_coin_ids([coin_name])
 
             else:
                 raise RuntimeError("All cases already handled")  # Logic error, all cases handled
+
+        # bulk-lookup the parent coin info for all coins we're about to add
+        parent_coins: List[bytes32] = [c.coin.parent_coin_info for c in deferred_coins_added]
+        parent_records: List[Optional[WalletCoinRecord]] = await self.coin_store.get_coin_records(parent_coins)
+
+        for dca, parent_record in zip(deferred_coins_added, parent_records):
```

Comment:
that's a good point. I should ensure that case is covered by a test

Reply:
Another example:
Coin state X creates coin A.
Coin state Y spends coin A.

X must be processed in the DB for A to exist, so that Y can be processed.